core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -127,7 +128,14 @@ private[deploy] class Master(
     logInfo(s"Running Spark version ${org.apache.spark.SPARK_VERSION}")
     webUi = new MasterWebUI(this, webUiPort)
     webUi.bind()
-    masterWebUiUrl = "http://" + masterPublicAddress + ":" + webUi.boundPort
+    if (reverseProxy) {
+      masterWebUiUrl = conf.get("spark.ui.reverseProxyUrl", null)
```

Comment:
Is there a reason we need this config? Would there ever be a case where the proxy URL is different from the master URL? Also from my testing this config is completely ignored (as long as it exists)